.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "openzeppelin-contracts"]
+	path = openzeppelin-contracts
+	url = https://github.com/OpenZeppelin/openzeppelin-contracts
```

Comment:
I am not clear on where we are locking to a package version. We should be using this tag (https://github.com/OpenZeppelin/openzeppelin-contracts/tree/v4.4.0) since we bumped the solidity version.  I also think we are adding complexity here by having 2 different ways we are now managing dependencies. 

It also breaks the unit tests.  I don't see much benefit to this and would prefer we revert back, simply by modifying


```
dependencies:
  - OpenZeppelin/openzeppelin-contracts@4.4.0

# path remapping to support imports from GitHub/NPM
compiler:
  solc:
    version: 0.8.10
    remappings:
      - "@openzeppelin=OpenZeppelin/openzeppelin-contracts@4.4.0"
    ```

Reply:
Good catch on the OZ version, should have bumped that too. The lock to the specific version comes from the use of git submodules; the `openzeppelin-contracts` subfolder will check out to the specific commit hash we're locking to.

I made the change to submodules because brownie's dependency specification is somewhat ambiguous. I'm not 100% sure where it gets the code from (which might be a me problem, but is certainly a bit scary), but `OpenZeppelin/openzeppelin-contracts` isn't the NPM `@openzeppelin/contracts` package, and if it's a reference to a tag on the github repo that's not an effective version lock because tags can be changed. If it were an NPM package, the lockfile would take care of that, but IDK what the brownie equivalent would be for that.

Reply:
OZ version bumped

Reply:
Further investigation shows that brownie just doesn't _do_ anything like a lockfile; that's probably an artifact of its pythonic heritage. Properly locked deps are absolutely essential for an auditable smart contract, so I've brought in the yearn-vaults repo via submodule as well, and updated conftest.py and the brownie config to remove the "brownie package manager" stuff and its associated remappings.

Reply:
Okay, please update the readme appropriately for the new dependency management with sub modules.

Reply:
Done!